File: ReservationProject/src/com/controllers/CTRLgererSalles.java
package com.controllers;

import java.io.IOException;
import java.io.PrintWriter;
import java.util.List;

import javax.servlet.ServletException;
import javax.servlet.annotation.WebServlet;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

import com.model.beans.Equipment;
import com.model.beans.Equipments;
import com.model.beans.Equipmentsalle;
import com.model.beans.Equipmentsalles;
import com.model.beans.Salle;
import com.model.beans.Salles;

/**
 * Servlet implementation class CTRLgererSalles
 */
@WebServlet({ "/CTRLgererSalles", "/gerersalles" })
public class CTRLgererSalles extends HttpServlet {
	private static final long serialVersionUID = 1L;

	/**
	 * @see HttpServlet#HttpServlet()
	 */
	public CTRLgererSalles() {
		super();
	}

	/**
	 * @see HttpServlet#doGet(HttpServletRequest request, HttpServletResponse
	 *      response)
	 */
	protected void doGet(HttpServletRequest request, HttpServletResponse response)
			throws ServletException, IOException {
		request.getRequestDispatcher("/WEB-INF/gererSalles.jsp").forward(request, response);
	}

	/**
	 * @see HttpServlet#doPost(HttpServletRequest request, HttpServletResponse
	 *      response)
	 */
	protected void doPost(HttpServletRequest request, HttpServletResponse response)
			throws ServletException, IOException {
		String choix = request.getParameter("choix");
		response.setContentType("text/plain");
		PrintWriter out = response.getWriter();
		boolean session =false;
		try {
		 session =	(boolean)request.getSession().getAttribute("session");
		} catch (Exception e) {
		
		}
		if(session){
			
		switch (choix) {
		// consulter salles
		case "ConsulterListSalles":

			System.out.println("** ConsulterListSalles  **");
			//
			Salles csalles = new Salles();
			//
			csalles = csalles.initSalles();
			List<Salle> clist = csalles.getListSalles();

			if (clist != null) {
				for (Salle cs : clist) {
					// System.out.println("Num" + cs.getNum());
					out.print("<tr id ='" + cs.getNum() + "'>" + "<td>" + cs.getNum() + "</td>" + "<td>"
							+ cs.getCapacite() + "</td>");
					out.print("<td><ul class='list-group'>");

					List<Equipment> seqlist = cs.getEquipments();
					if (seqlist != null) {
						out.print("<div class='panel-group'> <div class='panel panel-info'><div class='panel-heading'>"
								+ "<h4 class='panel-title'><a data-toggle='collapse' href='#collapse1" + cs.getNum()
								+ "'><span class='glyphicon glyphicon-chevron-down'></span>&nbsp;&nbsp;Numbre d Equipment de salle "
								+ cs.getNum() + "</a>" + "</h4></div> <div id='collapse1" + cs.getNum()
								+ "' class='panel-collapse collapse'>");

						for (Equipment eqs : seqlist) {
							out.print("<li class='list-group-item'><span class='badge'>"
									+ (new Equipmentsalles()).getNumbreEquipment(cs.getNum(), eqs.getId()) + "</span>"
									+ eqs.getNom() + "</li>");
						}
						if (csalles.verifierAccupation(cs.getNum()) == 1) {
							out.print("</ul>   </div> </div> </div></div></td>" + "<td class='text-center'>"
									+ "<a href='#' class='btn btn-info btn-md disabled' "
									+ "> <span class='glyphicon glyphicon-info'></span>deja Reserver" + "</a></td>"
									+ "</tr>");
						} else {
							out.print("</ul>   </div> </div> </div></div></td>" + "<td class='text-center'>"
									+ "<a href='#' class='btn btn-default btn-md disabled' "
									+ "> <span class='glyphicon glyphicon-info'></span>Salle Libre" + "</a></td>"
									+ "</tr>");
						}

					}
				}

			} else {
				out.println("vide");
			}

			break;

		/* Search Salles a Supprimer */
		case "SearchSalle":
			Salles SearchSS = new Salles();
			// List<Salle>
			try {
				int num = Integer.parseInt(request.getParameter("word"));
				SearchSS = SearchSS.SearchSalles(num);
				List<Salle> listSearchSS = SearchSS.getListSalles();
				if (listSearchSS != null) {
					for (Salle css : listSearchSS) {
						// System.out.println("Num" + css.getNum());
						out.print("<tr id='supptr" + css.getNum() + "'>" + "<td>" + css.getNum() + "</td>" + "<td>"
								+ css.getCapacite() + "</td>");
						out.print("<td><ul class='list-group'>");
						List<Equipment> seqlist = css.getEquipments();
						if (seqlist != null) {
							out.print(
									"<div class='panel-group'> <div class='panel panel-info'><div class='panel-heading'>"
											+ "<h4 class='panel-title'><a data-toggle='collapse' href='#collapse1"
											+ (css.getNum() + 1)
											+ "'><span class='glyphicon glyphicon-chevron-down'></span> &nbsp;&nbsp;list Equipment </a>"
											+ "</h4></div> <div id='collapse1" + (css.getNum() + 1)
											+ "' class='panel-collapse collapse'>");
							for (Equipment eqs : seqlist) {
								out.print("<li class='list-group-item'><span class='badge'>"
										+ (new Equipmentsalles()).getNumbreEquipment(css.getNum(), eqs.getId())
										+ "</span>" + eqs.getNom() + "</li>");
							}
							// System.out.println(SearchSS.verifierAccupation(css.getNum()));
							if (SearchSS.verifierAccupation(css.getNum()) == 1) {
								out.print("</ul>   </div> </div> </div></div></td>" + "<td class='text-center'>"
										+ "<a href='#' class='btn btn-danger btn-md disabled' "
										+ "> <span class='glyphicon glyphicon-remove'></span>Supprimer" + "</a></td>"
										+ "</tr>");
							} else {
								out.print("</ul>   </div> </div> </div></div></td>" + "<td class='text-center'>"
										+ "<a href='#' class='btn btn-danger btn-md'  onclick='ConfirmSalle("
										+ css.getNum() + ");'"
										+ "> <span class='glyphicon glyphicon-remove '></span>Supprimer" + "</a></td>"
										+ "</tr>");
							}
						} else {
							out.print("Vide");
						}
					}
				}

			} catch (Exception e) {
				out.print("<tr><td colspan='4'><div class='alert alert-danger alert-dismissable fade in' >"
						+ "<a href='#' class='close' data-dismiss='alert' aria-label='close'>&times;</a>"
						+ "<strong>Error : </strong> Entrer num de salle correct!</div></td></tr>");
			}
			break;
		/* supprimer salle */
		case "supprimersalle":
			System.out.println("** SupprimerSalle  **");
			
			try {
				Salles suppalles = new Salles();
				int idsalle = Integer.parseInt(request.getParameter("idsalle"));
				int res = suppalles.deleteSalle(idsalle);
				System.out.println("REs " + res);
				if (res == 1) {
					out.print("success");
				} else {
					out.print("failed");
				}
			} catch (Exception e) {
				out.print("Exception failed");
				System.out.println(e.getMessage());
			}
			break;
		/* ajouter salle */
		case "ajoutersalle":
			try {
				int nums = Integer.parseInt(request.getParameter("num"));
				int caps = Integer.parseInt(request.getParameter("capacite"));
				Salles ajsalle = new Salles();
				int resaj = ajsalle.insertSalle(new Salle(nums, caps));
				System.out.println(resaj+"++++++++++");
				if (resaj == 1) {
					out.print("1");
				} 
				if (resaj == -1) {
					out.print("-1");
				} 
				if (resaj == 0) {
					out.print("0");
				}
			} catch (Exception e) {
				out.print("Contriller Exception");
				System.out.println(e.getMessage());
			}
			break;
		case "SearchSalleAjEq":
			Salles SearchAjEq = new Salles();
			// List<Salle>
			try {
				int num = Integer.parseInt(request.getParameter("word"));
				SearchSS = SearchAjEq.SearchSalles(num);
				List<Salle> listSearchSS = SearchSS.getListSalles();
				if (listSearchSS != null) {
					for (Salle css : listSearchSS) {
						// System.out.println("Num" + css.getNum());
						out.print("<tr id='supptr" + css.getNum() + "'>" + "<td>" + css.getNum() + "</td>" + "<td>"
								+ css.getCapacite() + "</td>");
						out.print("<td><ul class='list-group'>");
						List<Equipment> seqlist = css.getEquipments();
						if (seqlist != null) {
							out.print(
									"<div class='panel-group'> <div class='panel panel-info'><div class='panel-heading'>"
											+ "<h4 class='panel-title'><a data-toggle='collapse' href='#collapse1"
											+ (css.getNum() + 1)
											+ "'><span class='glyphicon glyphicon-chevron-down'></span> &nbsp;&nbsp;list Equipment </a>"
											+ "</h4></div> <div id='collapse1" + (css.getNum() + 1)
											+ "' class='panel-collapse collapse in'>");
							for (Equipment eqs : seqlist) {
								out.print("<li class='list-group-item'><span class='badge'>"
										+ (new Equipmentsalles()).getNumbreEquipment(css.getNum(), eqs.getId())
										+ "</span>" + eqs.getNom() + "</li>");
							}
							// System.out.println(SearchSS.verifierAccupation(css.getNum()));
							if (SearchSS.verifierAccupation(css.getNum()) == 1) {
								out.print("</ul>   </div> </div> </div></div></td>" + "<td class='text-center'>"
										+ "<a href='#' class='btn btn-info btn-md disabled' "
										+ "> <span class='glyphicon glyphicon-add'></span>Aj-Equipment" + "</a></td>"
										+ "</tr>");
							} else {
								out.print("</ul>   </div> </div> </div></div></td>" + "<td class='text-center'>"
										+ "<a href='#' class='btn btn-warning btn-md'  data-toggle='modal' data-target='#myModalAddEq' "
										+ "onclick='setSalleID("+css.getNum()+")'"
										+ "> <span class='glyphicon glyphicon-add '></span>Aj-Equipment" + "</a></td>"
										+ "</tr>");
							}
						} else {
							out.print("Vide");
						}
					}
				}

			} catch (Exception e) {
				out.print("<tr><td colspan='4'><div class='alert alert-danger alert-dismissable fade in' >"
						+ "<a href='#' class='close' data-dismiss='alert' aria-label='close'>&times;</a>"
						+ "<strong>Error : </strong> Entrer num de salle correct!</div></td></tr>");
			}

			break;
			/* Get List Of Componenet */
		case  "GetListequipment" :
			
			Equipments equis =new Equipments();
			equis.initEquipments();
			List<Equipment> listeq = equis.getListEquipments();
			if (listeq != null ){
				for (Equipment e : listeq){
					out.print("	<option value='"+e.getId()+"'>"+e.getNom()+"</option>");
				}
			}else {
				out.print("Internal Error");
			}
			break;
			/* Modifier ou Ajouter equipment de la salle */
			
		case "AjouterEquipmnetSalle":
			System.out.println("+++++ AjouterEquipmnetSalle+++++++");
				int numbereq=Integer.parseInt(request.getParameter("numbereq"));
				int typeequipment = Integer.parseInt(request.getParameter("typeequipment"));
				int idsalle =Integer.parseInt(request.getParameter("id_salle"));
				  Equipmentsalles eqs =new Equipmentsalles();
				  int res=eqs.updateEquipmentsalle(new Equipmentsalle(idsalle, typeequipment, numbereq));
				  if (res == 1){
					  out.print("1");
				  }else {
					  out.print("0");	  
				  }
			
			break;
		/* Get Sall By Num */
		case "getSalle":
			Salles gsalle = new Salles();
			try {
				int num = Integer.parseInt(request.getParameter("word"));
				SearchSS = gsalle.SearchSalles(num);
				List<Salle> listSearchSS = SearchSS.getListSalles();
				if (listSearchSS != null) {
					for (Salle css : listSearchSS) {
						out.print("<tr id='supptr" + css.getNum() + "'>" + "<td>" + css.getNum() + "</td>" + "<td>"
								+ css.getCapacite() + "</td>");
						out.print("<td><ul class='list-group'>");
						List<Equipment> seqlist = css.getEquipments();
						if (seqlist != null) {
							out.print(
									"<div class='panel-group'> <div class='panel panel-info'><div class='panel-heading'>"
											+ "<h4 class='panel-title'><a data-toggle='collapse' href='#collapse1"
											+ (css.getNum() + 1)
											+ "'><span class='glyphicon glyphicon-chevron-down'></span> &nbsp;&nbsp;list Equipment </a>"
											+ "</h4></div> <div id='collapse1" + (css.getNum() + 1)
											+ "' class='panel-collapse collapse in'>");
							for (Equipment eqss : seqlist) {
								out.print("<li class='list-group-item'><span class='badge'>"
										+ (new Equipmentsalles()).getNumbreEquipment(css.getNum(), eqss.getId())
										+ "</span>" + eqss.getNom() + "</li>");
							}
							// System.out.println(SearchSS.verifierAccupation(css.getNum()));
							if (SearchSS.verifierAccupation(css.getNum()) == 1) {
								out.print("</ul>   </div> </div> </div></div></td>" + "<td class='text-center'>"
										+ "<a href='#' class='btn btn-info btn-md disabled' "
										+ "> <span class='glyphicon glyphicon-add'></span>Aj-Equipment" + "</a></td>"
										+ "</tr>");
							} else {
								out.print("</ul>   </div> </div> </div></div></td>" + "<td class='text-center'>"
										+ "<a href='#' class='btn btn-warning btn-md'  data-toggle='modal' data-target='#myModalAddEq' "
										+ "onclick='setSalleID("+css.getNum()+")'"
										+ "> <span class='glyphicon glyphicon-add '></span>Aj-Equipment" + "</a></td>"
										+ "</tr>");
							}
						} else {
							out.print("Vide");
						}
					}
				}

			} catch (Exception e) {
				out.print("<tr><td colspan='4'><div class='alert alert-danger alert-dismissable fade in' >"
						+ "<a href='#' class='close' data-dismiss='alert' aria-label='close'>&times;</a>"
						+ "<strong>Error : </strong> Entrer num de salle correct!</div></td></tr>");
			}

			break;

		default:
			break;
		}
	}else{
		 out.print("sessionfalse");
	}
	}
}

File: ReservationProject/WebContent/js/CTRLGererSalles.js
/**
 * 
 */
$(document).ready(function(){
	GetListEquipments();
	$("#searchbuttomsupp").click(function(){
		 $("#resultsearchSalle").hide();
	     SearchSalle();
	     $("#resultsearchSalle").fadeIn(2000);
    });
//	$("#Numse").click(function(){
//	
//    });
	$(document).on('click','#Numse',function(e){
		   e.preventDefault();
		   console.log('1');
		      //Code 
			AjouterEquipmnetSalle(); console.log('3');
		   });
	//Ajouterequip
	$("#searchajeq").click(function(){
		SearchSalleAjEq();
    });
	$("#AjuoterSalles").click(function(){
		AjouterSalle();
    });
}); 
function ConsulterListSalles() {

	$.ajax({
				url : 'gerersalles',
				data : {
					choix : 'ConsulterListSalles',
				},
				type : 'post',
				cache : false,
				success : function(data) {
				// alert(data);
					$('#ConsulterListSalles').html(data);
				},

				error : function() {
					$('#ConsulterListSalles')
							.html(
									"<div class='alert alert-danger alert-dismissable fade in' >"
											+ "<a href='#' class='close' data-dismiss='alert' aria-label='close'>&times;</a>"
											+ "<strong>Error : </strong> Server Not Responed !</div>");

				}
			});
}

// Search Salle a supprimer
function SearchSalle() {
	
	var word = document.getElementById('searchSS').value;

	$
			.ajax({
				url : 'gerersalles',
				data : {
					choix : 'SearchSalle',
					word : word
				},
				type : 'post',
				cache : false,
				success : function(data) {
				// alert(data);
					$('#resultsearchSalle').html(data);
				},
				error : function() {
					$('#resultsearchSalle').html(
									"<div class='alert alert-danger alert-dismissable fade in' >"
											+ "<a href='#' class='close' data-dismiss='alert' aria-label='close'>&times;</a>"
											+ "<strong>Error : </strong> Server Not Responed !</div>");
				}
			});
}

/* Supprimer Salle */

function ConfirmSalle(id){
	swal({
		  title: "Are you sure?",
		  text: "Once deleted, you will not be able to recover this imaginary file!",
		  icon: "warning",
		  buttons: true,
		  dangerMode: true,
		})
		.then((willDelete) => {
		  if (willDelete) {
			  
			  SupprimerSalle(id);
			  
		    swal("Poof! Your  Room has been deleted!", {
		      icon: "success",
		    });
		  } else {
		    swal("Your imaginary file is safe!");
		  }
		});
}
/**
 * Supprimer Salle
 */

function SupprimerSalle(id) {
	var idsalle = id;
		$.ajax({
					url  : 'gerersalles',
					data : {
						choix : 'supprimersalle',
						idsalle : idsalle
					},
					type : 'post',
					cache : false,
					success : function(data) {
						if(data == "success")
					    document.getElementById('supptr'+id).style.display = 'none';
					},
					error : function() {
						swal({
							  text: "internal Error !",
							  icon: "warning",
							  button: true,
							  dangerMode: true,
							});
					}
				});
}


/* Ajouter salle */

function AjouterSalle() {
	var num = document.getElementById('nums');
	var capacite = document.getElementById('capsts');
		$.ajax({
					url  : 'gerersalles',
					data : {
						choix : 'ajoutersalle',
						num : num.value,
						capacite: capacite.value
					},
					type : 'post',
					cache : false,
					success : function(data) {
						switch(data){
						case '1' :
						    document.getElementById('nums').value='';
							document.getElementById('capsts').value='';
							swal("Success!", "Success Ajoute!", "success");

							break;
						case '-1' : 
							swal("warning!",
									"Salle Deja Exist!!",
									"warning");	
							break;
							
						case '0' : 
							swal("danger!",
									"Exception Internal Error!",
									"danger");	
							break;
							
						}
						
					},
					error : function() {
						swal({
							  text: "Internal Error !",
							  icon: "warning",
							  button: true,
							  dangerMode: true,
							});
					}
				});
}
/* Seach salle to add component */

// Search Salle
function SearchSalleAjEq() {
	
	var word = document.getElementById('searchsajeq').value;

	$.ajax({
				url : 'gerersalles',
				data : {
					choix : 'SearchSalleAjEq',
					word : word
				},
				type : 'post',
				cache : false,
				success : function(data) {
				// alert(data);
					$('#resultsearchsajeq').html(data);
				},
				error : function() {
					$('#resultsearchsajeq').html(
									"<div class='alert alert-danger alert-dismissable fade in' >"
											+ "<a href='#' class='close' data-dismiss='alert' aria-label='close'>&times;</a>"
											+ "<strong>Error : </strong> Server Not Responed !</div>");
				}
			});
}

/* Ajouter Equipment a La salle */

function AjouterEquipmnetSalle(){
	  console.log('excutefunction');
	  
	var id_salle = document.getElementById('id_salle').value;
	var numbereq = document.getElementById('numbereq').value;
	var meunselection = document.getElementById("idequipment");
	var typeequipment = meunselection.options[meunselection.selectedIndex].value;
    console.log('AjouterEquipmnetSalle type idequip=> '+typeequipment);
    console.log('AjouterEquipmnetSalle numeroequi=>  ' +numbereq);
    console.log('AjouterEquipmnetSall id_sallee=>  '+id_salle);
	
    $.ajax({
		url : 'gerersalles',
		data : {
			choix : 'AjouterEquipmnetSalle',
			numbereq: numbereq,
			typeequipment: typeequipment,
			id_salle:id_salle
		},
		type : 'post',
		cache : false,
		success : function(data) {
			$('#myModalAddEq').modal('hide'); 
		//	document.getElementById('myModalAddEq').style.display='none';
			swal("Success!", "Success Ajoute Equipment!"+data, "success");
			GetSalle(id_salle);
		},
		error : function() {
			swal("danger!",
				"Exception Internal Error!",
					"danger");
		}
	});
//	$.ajax({
//					url  : 'gerersalles',
//					data : {
//						choix : 'AjouterEquipmnetSalle',
//						numbereq : numbereq,
//						typeequipment: typeequipment,
//						id: id
//					},
//					type : 'post',
//					cache : false,
//					success : function(data) {
//						switch(data){
//						case '1' :
//						   
//							swal("Success!", "Success Ajoute Equipment!", "success");
//							GetSalle(id);
//							break;
//						case '-1' : 
//							swal("warning!",
//									"Error Deja Exist!!",
//									"warning");	
//							break;
//							
//						case '0' : 
//							swal("danger!",
//									"Exception Internal Error!",
//									"danger");	
//							break;
//							
//						}
//						
//					},
//					error : function() {
//						swal({
//							  text: "Internal Error !",
//							  icon: "warning",
//							  button: true,
//							  dangerMode: true,
//							});
//					}
//				});
	
}
function setSalleID(id)
{
	console.log(id);
document.getElementById("id_salle").value= id;
}
/* Get Lsit of Component */

function GetListEquipments() {

	$.ajax({
				url : 'gerersalles',
				data : {
					choix : 'GetListequipment',
				},
				type : 'post',
				cache : false,
				success : function(data) {
					$('#idequipment').html(data);
				},
	error : function() {
					$('#idequipment')
							.html(
									"<div class='alert alert-danger alert-dismissable fade in' >"
											+ "<a href='#' class='close' data-dismiss='alert' aria-label='close'>&times;</a>"
											+ "<strong>Error : </strong> Server Not Responed !</div>");

				}
			});
}

/* Get Salle apres ajoute Equipment */

function GetSalle(id) {
	
	var word = id;

	$.ajax({
				url : 'gerersalles',
				data : {
					choix : 'getSalle',
					word : word
				},
				type : 'post',
				cache : false,
				success : function(data) {
				// alert(data);
					$('#resultsearchsajeq').html(data);
				},
				error : function() {
					$('#resultsearchsajeq').html(
									"<div class='alert alert-danger alert-dismissable fade in' >"
											+ "<a href='#' class='close' data-dismiss='alert' aria-label='close'>&times;</a>"
											+ "<strong>Error : </strong> Server Not Responed !</div>");
				}
			});
}
